reactor-support/src/test/java/tech/picnic/reactor/ReactorSpring4UtilTest.java

```diff
@@ -0,0 +1,232 @@
+        response.getContentAsString());
+  }
+
+  @Test
+  void testPublisherToSseWithKeepAliveAndError() throws Exception {
+    testScheduler.advanceTimeTo(Instant.EPOCH);
+    MockHttpServletResponse response =
+        doGet("/publisherToSse/with-keep-alive?value=error&repeat=1&interval=150")
+            .andReturn()
+            .getResponse();
+    testScheduler.advanceTimeTo(Instant.EPOCH.plus(Duration.ofMillis(149)));
+    assertEquals("data:keep-alive #0\n\n", response.getContentAsString());
+    testScheduler.advanceTimeTo(Instant.EPOCH.plus(Duration.ofMillis(200)));
+    assertEquals("data:keep-alive #0\n\n", response.getContentAsString());
+
+    // XXX: An error prevents further emissions, but there is no other evidence of it in the output.
+  }
+
+  @Test
+  void testPublisherToSseWithComplexObject() throws Exception {
+    verifySyncGetRequest(
+        "/publisherToSse/with-complex-object?repeat=2",
+        "" + "data:{\"name\":\"foo\",\"age\":0}\n\n" + "data:{\"name\":\"foo\",\"age\":1}\n\n");
+    verifySyncGetRequest(
+        "/publisherToSse/with-complex-object?mediaType=application/xml&repeat=2",
+        ""
+            + "data:<Person><name>foo</name><age>0</age></Person>\n\n"
+            + "data:<Person><name>foo</name><age>1</age></Person>\n\n");
+  }
+
+  private void verifySyncGetRequest(String request, String expectedContent) throws Exception {
+    doGet(request)
+        .andExpect(MockMvcResultMatchers.status().is(HttpStatus.OK.value()))
+        .andExpect(MockMvcResultMatchers.content().string(expectedContent));
+  }
+
+  private void verifyAsyncGetRequest(
+      String request, HttpStatus expectedStatus, String expectedContent) throws Exception {
+    mockMvc
+        .perform(MockMvcRequestBuilders.asyncDispatch(doGet(request).andReturn()))
+        .andExpect(MockMvcResultMatchers.status().is(expectedStatus.value()))
+        .andExpect(MockMvcResultMatchers.content().string(expectedContent));
+  }
+
+  private ResultActions doGet(String request) throws Exception {
+    return mockMvc
+        .perform(MockMvcRequestBuilders.get(request))
+        .andExpect(MockMvcResultMatchers.request().asyncStarted());
+  }
+
+  static class Person {
+    private final String name;
+    private final int age;
+
+    @JsonCreator
+    Person(String name, int age) {
+      this.name = name;
+      this.age = age;
+    }
+
+    public String getName() {
+      return name;
+    }
+
+    public int getAge() {
+      return age;
+    }
+  }
+
+  @RestController
+  static class TestController {
+    private final VirtualTimeScheduler virtualTimeScheduler;
+
+    TestController(VirtualTimeScheduler virtualTimeScheduler) {
+      this.virtualTimeScheduler = virtualTimeScheduler;
+    }
+
+    @GetMapping("/publisherToDeferredResult")
+    public DeferredResult<ImmutableList<String>> withPublisherToDeferredResult(
+        @RequestParam String value, @RequestParam(defaultValue = "0") int repeat) {
```

Comment:
The default value of repeat here is 0 instead of 1. The repeat value in RxJava indicates how often things are emitted, so setting it to 0 means no emissions at all. In reactor this is really a replay, so setting it to 0 means things are emitted once.